packages/medusa/src/services/price-list.ts

```diff
@@ -253,9 +253,21 @@ class PriceListService extends BaseService {
     config: FindConfig<PriceList> = { skip: 0, take: 20 }
   ): Promise<[PriceList[], number]> {
     const priceListRepo = this.manager_.getCustomRepository(this.priceListRepo_)
-
-    const query = this.buildQuery_(selector, config)
-    return await priceListRepo.findAndCount(query)
+    const q = selector.q
+    const { relations, ...query } = this.buildQuery_(selector, config)
+
+    if (q) {
+      delete query.where.q
+      const resultsWithCount =
+        await priceListRepo.getFreeTextSearchResultsAndCount(
+          q,
+          query,
+          relations
+        )
+
+      return resultsWithCount
```

Comment:
`return await priceListRepo.getFreeTextSearchResultsAndCount(...)` the variable is redundant